specs/experimental/custom-gas-token.md

```diff
@@ -0,0 +1,471 @@
+# Custom Gas Token
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Constants](#constants)
+- [Properties of a Gas Paying Token](#properties-of-a-gas-paying-token)
+  - [Automated Validation](#automated-validation)
+- [Configuring the Gas Paying Token](#configuring-the-gas-paying-token)
+- [Contract Modifications](#contract-modifications)
+  - [OptimismPortal](#optimismportal)
+    - [`depositERC20Transaction`](#depositerc20transaction)
+      - [Function Arguments](#function-arguments)
+    - [`depositTransaction`](#deposittransaction)
+    - [`setGasPayingToken`](#setgaspayingtoken)
+  - [StandardBridge](#standardbridge)
+  - [CrossDomainMessenger](#crossdomainmessenger)
+  - [SystemConfig](#systemconfig)
+    - [initialize](#initialize)
+    - [Getters](#getters)
+  - [L1Block](#l1block)
+  - [WETH9](#weth9)
+- [User Flow](#user-flow)
+  - [When ETH is the Native Asset](#when-eth-is-the-native-asset)
+  - [When an ERC20 Token is the Native Asset](#when-an-erc20-token-is-the-native-asset)
+  - [Differences](#differences)
+- [Upgrade](#upgrade)
+  - [L1Block Deployment](#l1block-deployment)
+  - [L2CrossDomainMessenger Deployment](#l2crossdomainmessenger-deployment)
+  - [L2StandardBridge Deployment](#l2standardbridge-deployment)
+  - [L1Block Proxy Update](#l1block-proxy-update)
+  - [L2CrossDomainMessenger Proxy Update](#l2crossdomainmessenger-proxy-update)
+  - [L2StandardBridge Proxy Update](#l2standardbridge-proxy-update)
+- [Selection of `ETHER_TOKEN_ADDRESS`](#selection-of-ether_token_address)
+- [Standard Config](#standard-config)
+- [Security Considerations](#security-considerations)
```

Comment:
Not quite a security consideration, maybe fits somewhere else, but we probably should mention something about the fee params and how the transaction fees paid in the custom gas token likely won't cover the L1 posting costs

Reply:
what could this text be like? not sure what it could look like @mds1

Reply:
We could add a section above security considerations with text like so:

### Fees

The OP Stack natively charges fees in terms of ether due to the fee formula taking into account the basefee and blobbasefee. When a custom gas token is used, fees are paid in the custom gas token but the conversion rate to ether is not taken into account as part of the protocol. It is assumed that the fees will be configured by the chain operator such that the revenue earned in custom gas token can be swapped into ether to pay for posting the data to L1.

Reply:
That looks great to me, thank you @tynes!